docs: add v2023.2 release notes (#3068)
freifunk-gluon/gluon: docs/releases/v2023.2.rst

Gluon 2023.2
============

Important notes
---------------

> Upgrades to v2023.2 and later releases are only supported from releases v2021.1 and later.

Reviewer: ```suggestion
Upgrades to v2023.2 and later releases are only supported from releases v2022.1 and later.
```

At least because of the EFI Bug upgrades will only be supported from v2022.1.

- Wavlink

  - WS-WN572HP3 4G

ramips-mt76x8
~~~~~~~~~~~~~

- ASUS

  - RT-AX53U

- ZyXEL

  - WSM20

Removed hardware support
------------------------

ath79-generic
~~~~~~~~~~~~~

- TP-Link

  - Archer C60 (v1)
  - RE355
  - RE450 (v1)

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - RE305

Docker Container
----------------

The Gluon build-container is now published to the GitHub container registry.
The container contains all the tools required to build Gluon images from source.

See the (`container registry <https://github.com/freifunk-gluon/gluon/pkgs/container/gluon-build>`_) for more information.

Bugfixes
--------

- Fixed script failure when reconfiguring interface groups without an assigned role.
- Host tools used to be built twice on first compilation.

Minor changes
-------------

- D-Link DIR-825 B1 factory images are no longer built due to size constraints.
  Please use a recent OpenWrt 23.05 image for factory installation and install Gluon
  using sysupgrade.
- The robots.txt now prohibits crawling the status page.
- Changed the order in which Gluon installs packages into the OpenWrt build system
  to favor Gluon and site packages over upstream OpenWrt packages.

> Known issues

Reviewer 2: @skorpy2009 might be worth mentioning that 7520 and 7530 will be bricked if someone install Gluon v2023.x
https://github.com/freifunk-gluon/gluon/issues/3023

Reviewer: The Issue only exists for v2023.1.x according to the Issue. v2023.2.x will be branched off of master.
Generally, if it's known beforehand the device would be removed.

------------

* The integration of the BATMAN_V routing algorithm is incomplete.

  - Mesh neighbors don't appear on the status page. (`#1726 <https://github.com/freifunk-gluon/gluon/issues/1726>`_)
    Many tools have the BATMAN_IV metric hardcoded, these need to be updated to account for the new throughput
    metric.
  - Throughput values are not correctly acquired for different interface types.
    (`#1728 <https://github.com/freifunk-gluon/gluon/issues/1728>`_)
    This affects virtual interface types like bridges and VXLAN.

* Default TX power on many Ubiquiti devices is too high, correct offsets are unknown
  (`#94 <https://github.com/freifunk-gluon/gluon/issues/94>`_)

  Reducing the TX power in the Advanced Settings is recommended.

* In configurations without VXLAN, the MAC address of the WAN interface is modified even when Mesh-on-WAN is disabled
  (`#496 <https://github.com/freifunk-gluon/gluon/issues/496>`_)

  This may lead to issues in environments where a fixed MAC address is expected (like VMware when promiscuous mode is disallowed).

> * EFI only systems won't boot due to removed EFI support (introduced in v2023.1). This was necessary to work around a bug that
>   causes a config loss during direct upgrades from v2021.1.x to v2023.1.x with the *x86-64*, *x86-generic* and *x86-legacy* targets
>   (`#2967 <https://github.com/freifunk-gluon/gluon/issues/2967>`_).
> 
>   It is planned that the next major release will reintroduce EFI support.

Reviewer: ```suggestion
```

only applied to v2023.1.x. The support was never removed in master.

https://github.com/freifunk-gluon/gluon/blob/master/targets/x86.inc#L47

Reviewer: It should probably also be mentioned somewhere that EFI systems are now supported again.

> Gluon 2023.2

Reviewer 3: Mention the issue with Ubiquiti Flash and ask for help

Reviewer 4: @herbetom mentioned in the meetup that they have a device available for opening and fixing this. But it might still be useful to ask for further devices for testing the final fix.

GitHub Actions
~~~~~~~~~~~~~~

Gluon build-tests now run inside a Docker container built from the gluon-build Dockerfile of the same version.

Bugfixes
--------

- Fixed script failure when reconfiguring interface groups without an assigned role.
- Host tools used to be built twice on first compilation.

Major changes
-------------

This release is based on the newest OpenWrt 23.05 release branch.
It ships with Linux kernel 5.15, wireless-backports 6.1.24 and batman-adv 2023.1.

Minor changes
-------------

- D-Link DIR-825 B1 factory images are no longer built due to size constraints.
  Please use a recent OpenWrt 23.05 image for factory installation and install Gluon
  using sysupgrade.
- The robots.txt now prohibits crawling the status page.
- Changed the order in which Gluon installs packages into the OpenWrt build system
  to favor Gluon and site packages over upstream OpenWrt packages.
- If enough nodes are updated, the batman-adv multicast optimizations originally introduced in Gluon 2021.1 for link-local IPv6 multicast addresses
  will be applied within the domain to routable IPv6 multicast addresses.

> - Gluon now uses mbedtls instead of WolfSSL for hostapd and wpa-supplicant.

Reviewer: Maybe add some call to action that Communities should check if they include something like libustream-wolfssl and then switch to libustream-mbedtls.

https://github.com/freifunk-darmstadt/site-ffda/commit/980a919a58eb7c650bc0dac400e91586919e311f

Reviewer 2: While we are at it, maybe also do the same for `libustream-openssl`: https://github.com/freifunkMUC/site-ffm/pull/327/commits/9adfc3ca42bfe9293c35859f5edcb466970c746d

Gluon 2023.2
============

Important notes
---------------

Upgrades to v2023.2 and later releases are only supported from releases v2022.1 and later.
This is due to migrations that have been removed to simplify maintenance.

Deprecation of tunneldigger VPN
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Tunneldigger is set to be removed from core in the next Gluon release. It is recommended
to migrate to fastd or WireGuard. Alternatively, tunneldigger will be moved to the
community-packages repository and can be installed from there.

Added hardware support
----------------------

armsr
~~~~~

- ARM

>   - System Ready

Reviewer: ```suggestion
  - SystemReady 32-bit (EFI)
  - SystemReady 64-bit (EFI)
```

ath79-generic
~~~~~~~~~~~~~

- AVM

  - FRITZ!Repeater 1750E

- Sophos

  - AP100c
  - AP55c

- TP-Link

  - Archer C60 (v1)
  - EAP225-Outdoor v3
  - TP-Link TL-WR2543N/ND (v1)

ath79-mikrotik
~~~~~~~~~~~~~~

- MikroTik

  - wAPR-2nD (wAP R)

ipq40xx-generic
~~~~~~~~~~~~~~~

- ZTE

  - MF289F

mediatek-filogic
~~~~~~~~~~~~~~~~

- ASUS

  - TUF-AX4200

- Cudy

  - WR3000 (v1)

- GL.iNet

  - GL-MT3000

- NETGEAR

  - WAX220

- Ubiquiti

  - Unifi 6 Plus

- ZyXEL

  - NWA50AX Pro

mpc85xx-p1010
~~~~~~~~~~~~~

- Enterasys

  - WS-AP3715i

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - EAP615-Wall

- Wavlink

  - WS-WN572HP3 4G

ramips-mt76x8
~~~~~~~~~~~~~

- ASUS

  - RT-AX53U

- ZyXEL

  - WSM20

Removed hardware support
------------------------

ath79-generic
~~~~~~~~~~~~~

- TP-Link

  - Archer C60 (v1)
  - RE355
  - RE450 (v1)

- Ubiquiti

  - NanoBeam 5AC 19 (XC)
  - NanoBeam M5 (XW)
  - NanoStation Loco M2/M5 (XW)
  - NanoStation M2/M5 (XW)

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - RE305

Features
--------

TLS Support
~~~~~~~~~~~

> Sluon now supports TLS for libustream. Communities can include the new `tls` feature

Reviewer: ```suggestion
Gluon now supports TLS for libustream. Communities can include the new `tls` feature
```

Gluon 2023.2
============

Important notes
---------------

Upgrades to v2023.2 and later releases are only supported from releases v2022.1 and later.
This is due to migrations that have been removed to simplify maintenance.

> Deprecation of tunneldigger VPN

Reviewer: ```suggestion
Deprecation of Tunneldigger VPN
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Tunneldigger is set to be removed from core in the next Gluon release. It is recommended
to migrate to fastd or WireGuard. Alternatively, tunneldigger will be moved to the
community-packages repository and can be installed from there.

Site changes
------------

Image customization
~~~~~~~~~~~~~~~~~~~

`GLUON_FEATURES` as well as `GLUON_PACKAGES` have been replaced by a more flexible customization framework
based on a Lua DSL. Package and feature selection can be specified more granularly at both target and device level.

Additionally, this framework also allows communities to specify which devices should or should not be built.
For more information, see the :ref:`documentation <site-image-customization>`.

Added hardware support
----------------------

armsr-armv7
~~~~~~~~~~~

- ARM

  - SystemReady 32-bit (EFI)

armsr-armv8
~~~~~~~~~~~

- ARM

  - SystemReady 64-bit (EFI)

ath79-generic
~~~~~~~~~~~~~

- AVM

  - FRITZ!Repeater 1750E

- Sophos

>   - AP100c
>   - AP55c

Reviewer: ```suggestion
  - AP100
  - AP100c
  - AP55
  - AP55c
```

- TP-Link

  - Archer C60 (v1)
  - EAP225-Outdoor v3

>   - TP-Link TL-WR2543N/ND (v1)

Reviewer: ```suggestion
  - TL-WR2543N/ND (v1)
```

- Gluon now uses mbedtls instead of WolfSSL for hostapd and wpa-supplicant.

Known issues
------------

* The integration of the BATMAN_V routing algorithm is incomplete.

  - Mesh neighbors don't appear on the status page. (`#1726 <https://github.com/freifunk-gluon/gluon/issues/1726>`_)
    Many tools have the BATMAN_IV metric hardcoded, these need to be updated to account for the new throughput
    metric.
  - Throughput values are not correctly acquired for different interface types.
    (`#1728 <https://github.com/freifunk-gluon/gluon/issues/1728>`_)
    This affects virtual interface types like bridges and VXLAN.

* Default TX power on many Ubiquiti devices is too high, correct offsets are unknown
  (`#94 <https://github.com/freifunk-gluon/gluon/issues/94>`_)

  Reducing the TX power in the Advanced Settings is recommended.

* In configurations without VXLAN, the MAC address of the WAN interface is modified even when Mesh-on-WAN is disabled
  (`#496 <https://github.com/freifunk-gluon/gluon/issues/496>`_)

  This may lead to issues in environments where a fixed MAC address is expected (like VMware when promiscuous mode is disallowed).

> * Ubiquiti airMax devices do not unlock their flash wripte-protect. These boards have been disabled for this

Reviewer 5: Should be write-protect not wripte-protect

Gluon 2023.2
============

Important notes
---------------

Upgrades to v2023.2 and later releases are only supported from releases v2022.1 and later.
This is due to migrations that have been removed to simplify maintenance.

Deprecation of Tunneldigger VPN
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Tunneldigger is set to be removed from core in the next Gluon release. It is recommended
to migrate to fastd or WireGuard. Alternatively, tunneldigger will be moved to the
community-packages repository and can be installed from there.

Site changes
------------

Image customization
~~~~~~~~~~~~~~~~~~~

`GLUON_FEATURES` as well as `GLUON_PACKAGES` have been replaced by a more flexible customization framework
based on a Lua DSL. Package and feature selection can be specified more granularly at both target and device level.

Additionally, this framework also allows communities to specify which devices should or should not be built.
For more information, see the :ref:`documentation <site-image-customization>`.

Added hardware support
----------------------

armsr-armv7
~~~~~~~~~~~

- ARM

>   - SystemReady 32-bit (EFI)

Reviewer 3: I think we should stick to the separate targets. I'll adapt the Model name to add the arm prefix.

>   - SystemReady 32-bit (EFI)

Reviewer 6: Okay. I still think we should mention that this is primarily for virtualization.

armsr-armv8
~~~~~~~~~~~

- ARM

  - SystemReady 64-bit (EFI)

ath79-generic
~~~~~~~~~~~~~

- AVM

  - FRITZ!Repeater 1750E

- Sophos

  - AP100
  - AP100c
  - AP55
  - AP55c

- TP-Link

  - Archer C60 (v1)
  - EAP225-Outdoor v3
  - TL-WR2543N/ND (v1)

ath79-mikrotik
~~~~~~~~~~~~~~

- MikroTik

  - wAPR-2nD (wAP R)

ipq40xx-generic
~~~~~~~~~~~~~~~

- ZTE

  - MF289F

mediatek-filogic
~~~~~~~~~~~~~~~~

- ASUS

  - TUF-AX4200

- Cudy

  - WR3000 (v1)

- GL.iNet

  - GL-MT3000

- NETGEAR

  - WAX220

- Ubiquiti

  - Unifi 6 Plus

- ZyXEL

  - NWA50AX Pro

mpc85xx-p1010
~~~~~~~~~~~~~

- Enterasys

  - WS-AP3715i

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - EAP615-Wall

- Wavlink

  - WS-WN572HP3 4G

ramips-mt76x8
~~~~~~~~~~~~~

- ASUS

  - RT-AX53U

- ZyXEL

  - WSM20

Removed hardware support
------------------------

ath79-generic
~~~~~~~~~~~~~

- TP-Link

  - Archer C60 (v1)
  - RE355
  - RE450 (v1)

- Ubiquiti

  - NanoBeam 5AC 19 (XC)
  - NanoBeam M5 (XW)
  - NanoStation Loco M2/M5 (XW)

>   - NanoStation M2/M5 (XW)

Reviewer 6: Needs an explanation (footnote?) for devices that are removed temporarily because of critical issues, but that we expect to be added again when the issues have been solved.

Reviewer: It's in the Knwon Issues that they are missing and will be readded once the issue is resolved. Maybe we could only mention it there.

If nobody were to readd them we could still remove that known issue in a future release and make a "Removed hardware support" annoucement for them.

And if it's fixed we can remove the Known Issue.

Reviewer 3: Yes, It's a known issue for this release. I think this should be enough.

Reviewer 6: As removed hardware support is always a topic of controversy, I'd prefer to make this as obvious as possible.

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - RE305

Features
--------

TLS support
~~~~~~~~~~~

> Gluon now supports TLS for libustream. Communities can include the new `tls` feature
> which automatically installs the corresponding package as well as the ca-bundle.
> 
> Communities should switch to this feature instead of including the packages individually.
> Custom packages should switch to this dependency.

Reviewer 6: I think this section should be reworded, as it is unclear what the TLS support refers to, and I'd argue that Gluon has always "supported" TLS.

```suggestion
Gluon now provides HTTPS client support when the `tls` feature is included in the site
configuration, allowing nodes to establish encrypted connections to autoupdater mirrors,
opkg repositories and other HTTPS servers.

Existing site configurations that add libustream TLS packages should switch to the `tls`
feature instead, which will always include the recommended TLS implementation as well
as common CA certificates (`ca-bundle`).

Communities should switch to this feature instead of including the packages individually.
Custom packages should switch to depend on the `gluon-tls` package instead of referencing
specific libustream packages directly.
```
(this is written assuming that libustream TLS is only used for HTTPS client support, are there
other uses?)

Reviewer: the `should switch` sounds a bit repetitive, but i currently don't have a better wording

EFI images
~~~~~~~~~~

Gluon x86-64 images now support systems using EFI boot. The same images are still compatible
with legacy MBR boot methods.

Support for SQM QoS with fastd
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Gluon now supports SQM QoS with fastd using CAKE. SQM is automatically enabled with devices
offering at least 200MB of system memory when bandwith limitations are configured.

The SQM support can be activated by including the `mesh-vpn-sqm` feature in the site configuration.

Docker container
~~~~~~~~~~~~~~~~

The Gluon build-container is now published to the GitHub container registry.
The container contains all the tools required to build Gluon images from source.

See the (`container registry <https://github.com/freifunk-gluon/gluon/pkgs/container/gluon-build>`_) for more information.

GitHub actions
~~~~~~~~~~~~~~

> Gluon build-tests now run inside a Docker container built from the gluon-build Dockerfile of the same version.

Reviewer 6: ```suggestion
Gluon build tests now run inside a Docker container built from the gluon-build Dockerfile of the same version.
```

Bugfixes
--------

- Fixed script failure when reconfiguring interface groups without an assigned role.
- Host tools used to be built twice on first compilation.

Major changes
-------------

This release is based on the newest OpenWrt 23.05 release branch.

> It ships with Linux kernel 5.15, wireless-backports 6.1.24 and batman-adv 2023.1.

Reviewer 6: ```suggestion
It ships with Linux kernel 5.15.y, wireless-backports 6.1.24 and batman-adv 2023.1.
```

Gluon 2023.2
============

Important notes
---------------

Upgrades to v2023.2 and later releases are only supported from releases v2022.1 and later.
This is due to migrations that have been removed to simplify maintenance.

Deprecation of Tunneldigger VPN
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Tunneldigger is set to be removed from the Gluon base repository in the next Gluon release. It is recommended

Reviewer: ```suggestion
Tunneldigger is set to be removed from the Gluon base repository in the next major Gluon release. It is recommended
```

to migrate to fastd or WireGuard. Tunneldigger will be moved to the
community-packages repository and can be installed from there as an alternative.

Site changes
------------

Image customization
~~~~~~~~~~~~~~~~~~~

`GLUON_FEATURES` as well as `GLUON_PACKAGES` have been replaced by a more flexible customization framework
based on Lua. Package and feature selection can be specified more granularly at both target and device level.

Additionally, this framework also allows communities to specify which devices should or should not be built.

> For more information, see the :ref:`documentation <site-image-customization>`.

Reviewer 6: Should we also add a short example of old vs. new configuration directly to the release notes, like we did with other site changes in the past?

Reviewer 3: I think the reference to the documentation is enough. I was thinking about referencing community sites that already switched (like ffda and muc) to show practical examples. I see them as more valuable. What do you think?

Reviewer 6: I'd still prefer a simple before/after comparison of two examples in the release notes, and I've opened https://github.com/freifunk-gluon/gluon/pull/3127 to improve the referenced docs. Should I just push suggested improvements as new commits to this branch, if adding them as comments is inconvenient?

Added hardware support
----------------------

armsr-armv7
~~~~~~~~~~~

- Arm

  - Arm SystemReady 32-bit (EFI)

armsr-armv8
~~~~~~~~~~~

- Arm

  - Arm SystemReady 64-bit (EFI)

ath79-generic
~~~~~~~~~~~~~

- AVM

  - FRITZ!Repeater 1750E

- Sophos

  - AP100
  - AP100c
  - AP55
  - AP55c

- TP-Link

  - Archer C60 (v1)
  - EAP225-Outdoor v3
  - TL-WR2543N/ND (v1)

ath79-mikrotik
~~~~~~~~~~~~~~

- MikroTik

  - wAPR-2nD (wAP R)

ipq40xx-generic
~~~~~~~~~~~~~~~

- ZTE

  - MF289F

mediatek-filogic
~~~~~~~~~~~~~~~~

- ASUS

  - TUF-AX4200

- Cudy

  - WR3000 (v1)

- GL.iNet

  - GL-MT3000

- NETGEAR

  - WAX220

- Ubiquiti

  - Unifi 6 Plus

- ZyXEL

  - NWA50AX Pro

mpc85xx-p1010
~~~~~~~~~~~~~

- Enterasys

  - WS-AP3715i

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - EAP615-Wall

- Wavlink

  - WS-WN572HP3 4G

ramips-mt76x8
~~~~~~~~~~~~~

- ASUS

  - RT-AX53U

- ZyXEL

  - WSM20

Removed hardware support
------------------------

ath79-generic
~~~~~~~~~~~~~

- TP-Link

  - Archer C60 (v1)
  - RE355
  - RE450 (v1)

- Ubiquiti

  - NanoBeam 5AC 19 (XC)
  - NanoBeam M5 (XW)
  - NanoStation Loco M2/M5 (XW)

>   - NanoStation M2/M5 (XW)

Reviewer 3: No

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - RE305

Features
--------

TLS support
~~~~~~~~~~~

Gluon now provides HTTPS client support when the `tls` feature is included in the site
configuration, allowing nodes to establish encrypted connections to autoupdater mirrors,
opkg repositories and other HTTPS servers.

Existing site configurations that add libustream TLS packages should switch to the `tls`
feature instead, which will always include the recommended TLS implementation as well
as common CA certificates (`ca-bundle`).

> Communities should switch to this feature instead of including the packages individually.

Reviewer 6: This sentence is redundant now and should be removed (not sure if I missed it in my previous comment)

Reviewer 3: Fixed

~~~~~~~~~~

Gluon x86-64 images now support systems using EFI boot. The same images are still compatible
with legacy MBR boot methods.

Support for SQM QoS with fastd
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Gluon now supports SQM QoS with fastd using CAKE. SQM is automatically enabled with devices
offering at least 200MB of system memory when bandwith limitations are configured.

The SQM support can be activated by including the `mesh-vpn-sqm` feature in the site configuration.

Docker container
~~~~~~~~~~~~~~~~

The Gluon build-container is now published to the GitHub container registry.
The container contains all the tools required to build Gluon images from source.

See the (`container registry <https://github.com/freifunk-gluon/gluon/pkgs/container/gluon-build>`_) for more information.

GitHub actions
~~~~~~~~~~~~~~

Gluon build tests now run inside a Docker container built from the gluon-build Dockerfile of the same version.

Bugfixes
--------

- Fixed script failure when reconfiguring interface groups without an assigned role.
- Host tools used to be built twice on first compilation.

Major changes
-------------

This release is based on the newest OpenWrt 23.05 release branch.

> It ships with Linux kernel 5.15.y, wireless-backports 6.1.24 and batman-adv 2023.1.

Reviewer 7: shouldn't this state the actual version without a placeholder?

```suggestion
It ships with Linux kernel 5.15.139, wireless-backports 6.1.24 and batman-adv 2023.1.
```

Additionally, this framework also allows communities to specify which devices should or should not be built.
For more information, see the :ref:`documentation <site-image-customization>`.

Added hardware support
----------------------

armsr-armv7
~~~~~~~~~~~

- Arm

  - Arm SystemReady 32-bit (EFI)

armsr-armv8
~~~~~~~~~~~

- Arm

  - Arm SystemReady 64-bit (EFI)

ath79-generic
~~~~~~~~~~~~~

- AVM

  - FRITZ!Repeater 1750E

- Sophos

  - AP100
  - AP100c
  - AP55
  - AP55c

- TP-Link

  - Archer C60 (v1)
  - EAP225-Outdoor v3
  - TL-WR2543N/ND (v1)

ath79-mikrotik
~~~~~~~~~~~~~~

- MikroTik

  - wAPR-2nD (wAP R)

ipq40xx-generic
~~~~~~~~~~~~~~~

- ZTE

  - MF289F

mediatek-filogic
~~~~~~~~~~~~~~~~

- ASUS

  - TUF-AX4200

- Cudy

  - WR3000 (v1)

- GL.iNet

  - GL-MT3000

- NETGEAR

  - WAX220

- Ubiquiti

  - Unifi 6 Plus

- ZyXEL

  - NWA50AX Pro

mpc85xx-p1010
~~~~~~~~~~~~~

- Enterasys

  - WS-AP3715i

ramips-mt7621
~~~~~~~~~~~~~

- TP-Link

  - EAP615-Wall

- Wavlink

  - WS-WN572HP3 4G

ramips-mt76x8
~~~~~~~~~~~~~

- ASUS

  - RT-AX53U

- ZyXEL

  - WSM20

Removed hardware support
------------------------

> Ubiquiti airMax devices have been removed due to an unsolved issue with the flash write-protect.

Reviewer 6: ```suggestion
Ubiquiti airMax devices have been removed temporarily due to an unsolved issue with the flash write-protect.
```
